### PR TITLE
CB-10771: Fixing failure when empty string passed as a value for opti…

### PR DESCRIPTION
**Pull request by rakatyal**
…on parameter in upload function

CreateUploadAsync fails when passed an empty string as a value for SetText function. It also fails when we skip specifying a value for it. So when a user passes an empty string, skip the parameter instead.

### CODE REVIEW

**Comment by rakatyal** [comment]
@riknoll, @jasongin: Please review.

**Comment by riknoll** [comment]
Is this the only workaround? If so, this quirk should definitely be documented.

**Comment by rakatyal** [comment]
I believe so. As mentioned, passing empty string as a value for SetText function. It also fails when we skip specifying a value for it. One way to avoid this is leaving out this chunk, but then we lose sending the key for the option parameter. I can document it as a quirk.

File: src/windows/FileTransferProxy.js [hunk 1]
```diff
@@ -359,7 +359,13 @@ exec(win, fail, 'FileTransfer', 'upload',
                         if (params.hasOwnProperty(key)) {
                             var contentPart = new Windows.Networking.BackgroundTransfer.BackgroundTransferContentPart();
                             contentPart.setHeader("Content-Disposition", "form-data; name=\"" + key + "\"");
-                            contentPart.setText(params[key]);
+                            // CreateUploadSync fails when supplied with empty value for setText function. Pass " " instead when not specified.
+                            if (params[key]) {
+                                contentPart.setText(params[key]);
+                            }
+                            else {
+                                contentPart.setText(" ");
```
Comment (jasongin) [inline src/windows/FileTransferProxy.js]: Should a null/empty parameter just be omitted from the transfer instead?

Comment (rakatyal) [inline src/windows/FileTransferProxy.js]: That's an option. But we would lose sending the 'key' as well in that case. I thought this was the lesser of the two evils.

**Comment by rakatyal** [comment]
Updated to ignoring the parameter all together.

**Comment by paulhickman-a365** [comment]
Won't using "params[key]" as the test make it also ignore non-string values such as 0 or boolean false? Testing for undefined, null and "" explicitly would be better.

**Comment by rakatyal** [comment]
Yes it would but the options parameter is supposed to have only string key-value pairs as mentioned in the README.

**Comment by rakatyal** [comment]
I tested it on Android and apparently we do accept non string values and I don't see any reason why we should not. So your suggestion sounds fair to me.

**Comment by jasongin** [comment]
The update to check for an empty toString() result (as discussed) looks good.

**Comment by rakatyal** [comment]
Merged [here](https://github.com/apache/cordova-plugin-file-transfer/commit/584754869befd2d899f8e20270ba086ffe525625).
